Explicit trust relationships (and privacy considerations) (#342)
ietf-wg-privacypass/base-drafts: draft-ietf-privacypass-architecture.md

The Privacy Pass architecture consists of four logical entities --
Client, Origin, Issuer, and Attester -- that work in concert
for token redemption and issuance. This section presents an overview
of Privacy Pass, a high-level description of the threat model and

> privacy goals of the architecture, and the purpose of the the redemption
> and issuance protocols, including requirements on the relevant participants
> in these protocols, necessary to achieve the intended privacy goals.

Reviewer: ```suggestion
privacy goals of the architecture, and the goals and requirements of the
redemption and issuance protocols.
```

: The interactions and set of information shared between
the Client and Attester only, for the purposes of attesting the vailidity of
the Client. This context includes all information associated with attestation,
such as the timestamp of the event and any Client visible information,
including information needed for the attestation procedure to complete.

The privacy goals of Privacy Pass assume a threat model in which Origins trust
specific Issuers to produce tokens, and Issuers in turn trust one or more
Attesters to correctly run the attestation procedure with Clients. This
arrangement ensures that tokens which validate for a given Issuer were only

> issued to a Client that successfully complete attestation with an Attester that

Reviewer: ```suggestion
issued to a Client that successfully completed attestation with an Attester that
```

the Issuer trusts. Moreover, this arrangement means that if an Origin accepts

> tokens issued by an Issuer that trusts multiple Attesters then a Client can

Reviewer: ```suggestion
tokens issued by an Issuer that trusts multiple Attesters, then a Client can
```

> use any of these Attesters to issue and redeem tokens for the Origin.

Reviewer: ```suggestion
use any one of these Attesters to issue and redeem tokens for the Origin.
```

The mechanisms for establishing trust between each entity in this arrangement

> are deployment specific. For example, in settings where Attesters and Issuers

Reviewer: ```suggestion
are deployment-specific. For example, in settings where Attesters and Issuers
```

be issued and then spent multiple times in response to any such challenge.
See {{Section 2.1.1 of AUTHSCHEME}} for discussion.

How Clients respond to token challenges can have privacy implications.

> For example, if a token challenge admits a choice of Issuer, then the choice

Reviewer: What does it mean for a "token challenge to admit a choice of Issuer"?

Author: As a hypothetical, imagine a token challenge specified more than one issuer, or if clients were given two challenges for two different issuers.

Reviewer: I see. Our current challenge does not allow this to be expressed. How about:
```suggestion
For example, if an Origin allows the Client to choose an Issuer, then the choice
```

In this model, the Origin and Issuer are operated by the same entity, separate
from the Attester, as shown in the figure below. The Issuer accepts token
requests that come from trusted Attesters. Since the Attester and Issuer are
separate entities, this model requires some mechanism by which Issuers
establish trust in the Attester (as described in {{privacy-and-trust}}).
For example, in settings where the Attester is a Client-trusted service that
directly communicates with the Issuer, one way to establish this trust is via

> mutually-authenticated TLS. However, alernative authentication mechanisms are

Reviewer: ```suggestion
mutually-authenticated TLS. However, alternative authentication mechanisms are
```
